stdlib/public/core/Collection.swift

```diff
@@ -101,8 +101,10 @@ public struct IndexingGenerator<Elements : Indexable>
   ///
   /// - Requires: No preceding call to `self.next()` has returned `nil`.
   public mutating func next() -> Elements._Element? {
-    return _position == _elements.endIndex
-    ? .None : .Some(_elements[_position++])
+    guard _position != _elements.endIndex else { return .None }
```

Comment:
Wouldn't be `if` better and reverse the condition? The guard seems to be good for unwrapping but just for checking booleans it looks weird.